homeassistant/components/modbus/bit_switch.py

```diff
@@ -0,0 +1,201 @@
+            return
+
+        if isinstance(result, (ModbusException, ExceptionResponse)):
+            self._available = False
+            return
+
+        self._available = True
+        return int(result.registers[0])
+
+    def _write_register(self, address, value):
+        """Write holding register or coil using the Modbus hub slave."""
+        try:
+            self._hub.write_register(self._slave, address, value)
+        except ConnectionException:
+            self._available = False
+            return False
+
+        self._available = True
+        return True
+
+
+class ModbusRegisterBitSwitch(ModbusBaseSwitch, SwitchEntity):
+    """Representation of a Modbus register switch."""
+
+    def __init__(self, hub: ModbusHub, config: dict[str, Any]):
+        """Initialize the register switch."""
+        super().__init__(ModbusReadCache(hub), config)
+        self._register = config[CONF_ADDRESS]
+        self._verify_state = config[CONF_VERIFY_STATE]
+        self._verify_register = config.get(CONF_VERIFY_REGISTER, self._register)
+        self._register_type = config[CONF_INPUT_TYPE]
+
+        command_bit_numer = int(config[CONF_COMMAND_BIT_NUMBER])
+        status_bit_numer = int(config.get(CONF_STATUS_BIT_NUMBER, command_bit_numer))
+        assert 0 <= command_bit_numer < 16
+        assert 0 <= status_bit_numer < 16
+
+        self._command_bit_mask = 1 << command_bit_numer
+        self._status_bit_mask = 1 << status_bit_numer
+
+        self._is_on = None
+
+    def turn_on(self, **kwargs):
+        """Set switch on."""
+        # Only holding register is writable
+        if self._register_type != CALL_TYPE_REGISTER_HOLDING:
+            return
+        register_value = self._read_register(self._verify_register)
+        if register_value is None:
+            self._available = False
+            return
+        if self._write_register(
+            self._register, register_value | self._command_bit_mask
+        ):
+            self._is_on = True
+        self.schedule_update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Set switch off."""
+        # Only holding register is writable
+        if self._register_type != CALL_TYPE_REGISTER_HOLDING:
+            return
+        register_value = self._read_register(self._verify_register)
+        if register_value is None:
+            self._available = False
+            return
+        if self._write_register(
+            self._register, register_value & ~self._command_bit_mask
+        ):
+            self._is_on = False
+        self.schedule_update_ha_state()
+
+    def _update(self):
+        """Update the state of the switch."""
+        if not self._verify_state:
+            return
+        value = self._read_register(self._verify_register)
+        if value is not None:
+            self._is_on = bool(value & self._status_bit_mask)
+            self._available = True
+        else:
+            self._available = False
```

Comment:
But this is already done in _read...

Setting of _available seems duplicated in many places and inconsistent.